src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -411,31 +452,42 @@ module.exports = function(
       return false;
     }
 
-    // If we didn't bail out we need to construct a new instance. We don't
-    // want to reuse one that failed to fully mount.
-    const newInstance = constructClassInstance(workInProgress, newProps);
-    newInstance.props = newProps;
-    newInstance.state = newState = newInstance.state || null;
-    newInstance.context = newContext;
+    // componentWillMount may have called setState. Process the update queue.
```

Comment:
This comment is in the wrong place. Do you mean `componentWillReceiveProps`? If so, shouldn't this be gated on the existence of a `componentWillReceiveProps`?